protostar-rust/src/lib.rs

```diff
@@ -1,36 +1,64 @@
-use pyo3::exceptions::RuntimeError;
-use pyo3::prelude::*;
-use pyo3::wrap_pyfunction;
+use anyhow::{anyhow, Context, Result};
+use itertools::Itertools;
+use std::path::PathBuf;
+use walkdir::WalkDir;
 
-use anyhow::Context;
-use std::collections::HashMap;
-
-use cairo_lang_protostar::test_collector::collect_tests;
+use cairo_lang_protostar::test_collector::{collect_tests, LinkedLibrary};
 use cairo_lang_runner::{RunResultValue, SierraCasmRunner};
+use cairo_lang_sierra_to_casm::metadata::MetadataComputationConfig;
 
 fn run_result_value_to_string(run_result: RunResultValue) -> String {
-    match run_result {
+    return match run_result {
         RunResultValue::Success(data) => {
-            return format!("Success {:?}", data);
+            format!("Success {:?}", data)
         }
         RunResultValue::Panic(data) => {
-            return format!("Panic {:?}", data);
+            format!("Panic {:?}", data)
+        }
+    };
+}
+
+pub fn run_tests(input_path: PathBuf, linked_libraries: Option<Vec<LinkedLibrary>>) -> Result<()> {
+    for entry in WalkDir::new(&input_path) {
+        if entry.is_err() {
+            continue;
+        }
+
+        let entry = entry.unwrap();
+        let path = entry.path();
+
+        if path.is_file() && path.extension().map_or(false, |ex| ex == "cairo") {
+            run_tests_in_file(entry.path().to_path_buf(), linked_libraries.clone())?;
         }
     }
+    Ok(())
```

Comment:
Can we separate files collection from tests running?

Reply:
Sure